lib/charms/opensearch/v0/helper_networking.py

```diff
@@ -25,9 +26,22 @@
 logger = logging.getLogger(__name__)
 
 
-def get_host_ip(charm: CharmBase, peer_relation_name: str) -> str:
+class NetworkType(str, Enum):
+    """NetworkType Enum."""
+
+    PUBLIC = "public"
+    PRIVATE = "private"
+    BIND = "bind"
+
+
+def get_host_ip(charm: CharmBase, relation_name: str, type: NetworkType = NetworkType.BIND) -> str:
     """Fetches the IP address of the current unit."""
-    address = charm.model.get_binding(peer_relation_name).network.bind_address
+    if type == NetworkType.PRIVATE:
+        address = charm.model.get_binding(relation_name).network.private_address
```

Comment:
I do not find any mention of `private_address` in the `Network` class. Can you clarify where it's coming from?